ARCHITECTURE.md

```diff
@@ -0,0 +1,95 @@
+
+# Architecture
+
+This document should be updated semi-regularly. Feel free to open an issue if it hasn't been updated in more than a year.
+
+## Goals
+
+The major goal of Vello is to provide a high quality GPU accelerated renderer suitable for a range of 2D graphics applications, including rendering for GUI applications, creative tools, and scientific visualization.
+
+Vello emerges from being a research project, which attempts to answer these hypotheses:
+
+- To what extent is a compute-centered approach better than rasterization ([Direct2D])?
+- To what extent do "advanced" GPU features (subgroups, descriptor arrays, device-scoped barriers) help?
```

Comment:
I don't think the scare quotes are necessary. If the word doesn't seem nice then an alternative could be *modern*.
```suggestion
- To what extent do advanced GPU features (subgroups, descriptor arrays, device-scoped barriers) help?
```